// reakit/es/Flex.js
import styled from './styled.js';
import { a as as } from './as-3754c325.js';
import { a as bool$1 } from './chunk-3dd91a32.js';
import Box from './Box.js';
import { func, string, oneOfType, arrayOf, bool, number, oneOf, object, node } from 'prop-types';
import { theme } from 'styled-tools';
import 'styled-components';
import './chunk-fb907809.js';
import 'react';
import 'hoist-non-react-statics';
import '@emotion/is-prop-valid';
import './chunk-a7ce379e.js';

var directions = ["row", "column", "rowReverse", "columnReverse"];
var wraps = ["nowrap", "wrap", "wrapReverse"];
var Flex = styled(Box).withConfig({
  displayName: "Flex",
  componentId: "sp5dv1-0"
})(["display:flex;&&&{", ";", ";}", ";"], bool$1("flex-direction", directions), bool$1("flex-wrap", wraps), theme("Flex")); // @ts-ignore

Flex.propTypes = {
  row: bool,
  column: bool,
  rowReverse: bool,
  columnReverse: bool,
  nowrap: bool,
  wrap: bool,
  wrapReverse: bool
};
var Flex$1 = as("div")(Flex);

export default Flex$1;
