// reakit/lib/Backdrop.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('./styled.js');
var as = require('./as-84eb94f3.js');
var Hidden = require('./Hidden.js');
var styledTools = require('styled-tools');
require('styled-components');
require('./chunk-582bef72.js');
require('react');
require('hoist-non-react-statics');
require('@emotion/is-prop-valid');
require('./chunk-e9f69eed.js');
require('./chunk-f0c301b2.js');
require('./Box.js');
require('prop-types');
require('./chunk-25591e69.js');
require('./chunk-3027c8cd.js');
require('constate');
require('./chunk-0f087053.js');
require('react-dom');

var Backdrop = styled.default(Hidden.default).withConfig({
  displayName: "Backdrop",
  componentId: "sc-1mc2k5c-0"
})(["position:fixed;top:0;left:0;width:100%;height:100%;z-index:998;-webkit-tap-highlight-color:rgba(0,0,0,0);-moz-tap-highlight-color:rgba(0,0,0,0);", ";"], styledTools.theme("Backdrop"));
Backdrop.defaultProps = {
  role: "button",
  tabIndex: -1,
  opaque: true,
  palette: "shadow",
  tone: 2
};
var Backdrop$1 = as.as("div")(Backdrop);

exports.default = Backdrop$1;
