// reakit/lib/Box.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('./styled.js');
var __chunk_1 = require('./chunk-582bef72.js');
var as = require('./as-84eb94f3.js');
var __chunk_3 = require('./chunk-e9f69eed.js');
var PropTypes = require('prop-types');
var styledTools = require('styled-tools');
var React = require('react');
require('styled-components');
require('hoist-non-react-statics');
require('@emotion/is-prop-valid');
require('./chunk-f0c301b2.js');

var positions = ["static", "absolute", "fixed", "relative", "sticky"];

var BoxComponent = function BoxComponent(_ref) {
  var T = _ref.as,
      props = __chunk_1._objectWithoutPropertiesLoose(_ref, ["as"]);

  return React.createElement(T, props);
};

var Box = styled.default(BoxComponent).withConfig({
  displayName: "Box",
  componentId: "sc-1tbf5pj-0"
})(["margin:unset;padding:unset;border:unset;background:unset;font:unset;font-family:inherit;font-size:100%;box-sizing:border-box;background-color:", ";color:", ";&:focus:not(:focus-visible){outline:none;}", ";&&&{", ";}"], __chunk_3.bgColorWithProps, __chunk_3.textColorWithProps, styledTools.theme("Box"), __chunk_3.bool("position", positions));
var asTypes = [PropTypes.func, PropTypes.string]; // @ts-ignore

Box.propTypes = {
  as: PropTypes.oneOfType(asTypes.concat([PropTypes.arrayOf(PropTypes.oneOfType(asTypes))])),
  opaque: PropTypes.bool,
  palette: PropTypes.string,
  tone: PropTypes.number,
  static: PropTypes.bool,
  absolute: PropTypes.bool,
  fixed: PropTypes.bool,
  relative: PropTypes.bool,
  sticky: PropTypes.bool
};
var Box$1 = as.as("div")(Box);

exports.default = Box$1;
