// reakit/lib/Card.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('./styled.js');
var as = require('./as-84eb94f3.js');
var __chunk_2 = require('./chunk-f0c301b2.js');
var Box = require('./Box.js');
var __chunk_7 = require('./chunk-8931826f.js');
var PropTypes = require('prop-types');
var styledTools = require('styled-tools');
require('styled-components');
require('./chunk-582bef72.js');
require('react');
require('hoist-non-react-statics');
require('@emotion/is-prop-valid');
require('./chunk-e9f69eed.js');

var CardFit = styled.default(Box.default).withConfig({
  displayName: "CardFit",
  componentId: "sc-17ckqrf-0"
})(["", ";"], styledTools.theme("CardFit"));
var CardFit$1 = as.as("div")(CardFit);

var Card = styled.default(Box.default).withConfig({
  displayName: "Card",
  componentId: "sc-19jv5xv-0"
})(["display:inline-block;&& > *:not(", "){margin:", ";}", ";"], __chunk_7.getSelector(CardFit$1), styledTools.withProp("gutter", __chunk_2.numberToPx), styledTools.theme("Card")); // @ts-ignore

Card.propTypes = {
  gutter: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};
Card.defaultProps = {
  gutter: "1rem",
  opaque: true,
  palette: "background",
  tone: -1
};
var Card$1 = as.as("div")(Card);

var index = Object.assign(Card$1, {
  Fit: CardFit$1
});

exports.default = index;
