// reakit/lib/chunk-cecf6815.js
'use strict';

var styled = require('./styled.js');
var as = require('./as-84eb94f3.js');
var __chunk_5 = require('./chunk-3027c8cd.js');
var __chunk_6 = require('./chunk-0f087053.js');
var styledTools = require('styled-tools');

var OverlayContainer = __chunk_5.HiddenContainer;

var OverlayToggle = styled.default(__chunk_6.HiddenToggle).withConfig({
  displayName: "OverlayToggle",
  componentId: "sc-1u1lo2-0"
})(["", ";"], styledTools.theme("OverlayToggle"));
var OverlayToggle$1 = as.as("button")(OverlayToggle);

var OverlayShow = styled.default(__chunk_6.HiddenShow).withConfig({
  displayName: "OverlayShow",
  componentId: "c48b0n-0"
})(["", ";"], styledTools.theme("OverlayShow"));
var OverlayShow$1 = as.as("button")(OverlayShow);

var OverlayHide = styled.default(__chunk_6.HiddenHide).withConfig({
  displayName: "OverlayHide",
  componentId: "sc-1d97up5-0"
})(["", ";"], styledTools.theme("OverlayHide"));
var OverlayHide$1 = as.as("button")(OverlayHide);

exports.OverlayContainer = OverlayContainer;
exports.OverlayToggle = OverlayToggle$1;
exports.OverlayShow = OverlayShow$1;
exports.OverlayHide = OverlayHide$1;
