// reakit/lib/Flex.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('./styled.js');
var as = require('./as-84eb94f3.js');
var __chunk_3 = require('./chunk-e9f69eed.js');
var Box = require('./Box.js');
var PropTypes = require('prop-types');
var styledTools = require('styled-tools');
require('styled-components');
require('./chunk-582bef72.js');
require('react');
require('hoist-non-react-statics');
require('@emotion/is-prop-valid');
require('./chunk-f0c301b2.js');

var directions = ["row", "column", "rowReverse", "columnReverse"];
var wraps = ["nowrap", "wrap", "wrapReverse"];
var Flex = styled.default(Box.default).withConfig({
  displayName: "Flex",
  componentId: "sp5dv1-0"
})(["display:flex;&&&{", ";", ";}", ";"], __chunk_3.bool("flex-direction", directions), __chunk_3.bool("flex-wrap", wraps), styledTools.theme("Flex")); // @ts-ignore

Flex.propTypes = {
  row: PropTypes.bool,
  column: PropTypes.bool,
  rowReverse: PropTypes.bool,
  columnReverse: PropTypes.bool,
  nowrap: PropTypes.bool,
  wrap: PropTypes.bool,
  wrapReverse: PropTypes.bool
};
var Flex$1 = as.as("div")(Flex);

exports.default = Flex$1;
